src/main/java/io/aiven/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -1350,6 +1356,44 @@ public String buildInsertStatement(
         return builder.toString();
     }
 
+    @Override
+    public String buildMultiInsertStatement(final TableId table,
+                                            final int records,
+                                            final Collection<ColumnId> keyColumns,
+                                            final Collection<ColumnId> nonKeyColumns) {
+
+        if (records < 1) {
```

Comment:
Shouldn't we check the maximum number of records here? Does `Integer.MAX_INT` fit any database and all the constrains of the used expression builders and JDBC internals (in case if final expression string is build, for example)?

Reply:
@akudiyar For the reasons you mentioned above, I think we should have an upper limit on the number of records. I don't think it's possible to have it calculated precisely, because it appears the DBs differ a lot in the size of the query which can be sent to them (from what I could find you can send 1 GB to Pg, and in MySQL it's 64 MB by default). If the limit is too low, then users cannot use a large batch (if they want). So maybe something like 100k sounds good?

Reply:
I think it may be reasonable to have a setting somewhere or an environment option that can be adjusted by the user. And speaking about the size -- that's a good point that it should be measured not only by the count of records, but also by the size of the batch. In this case we can also have a setting and a default value of 64 MB for example, and count the size in the method that populates the query argument list. However even in this place the size of the query can easily reach several GBs (having 2 symbols Integer.MAX_VALUE times will be 4GB). Probably we need to have a query size limit as an option somewhere in the underlying query builder code for that.

Reply:
That are some good thoughts. Thinking once more about this, I have this question: what would be the use of those limits? We can either use them:
1. for warning a user (throwing an exception, i.e. no action done in the connector) or 
2. so that the connector automatically splits the batch into smaller batches based on the setting.

Option #2 may be a bit complex (because different DBs have different limits, and the single limit we would have in the connectors means that it would be least limit across all the DBs).

Option #1 I believe is good enough. In fact, I believe that even in the case we don't do anything, the driver will complain, so the user would still get warned (albeit, the message might not be that readable).

Reply:
I agree that the second option is better, if it's not very complex for implementing. The first one may disrupt the user workflow in an unexpected place.

Reply:
Sure. I'm not sure I can get to it that soon, so WDYT about having that in a separate issue/PR?

Reply:
I thank that creating a task looks reasonable for unblocking this patch.

Reply:
@akudiyar Here's the issue: https://github.com/aiven/jdbc-connector-for-apache-kafka/issues/117.